src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1684,7 +1684,7 @@ private static bool AccessingAutoPropertyFromConstructor(BoundExpression receive
             var propertyIsStatic = propertySymbol.IsStatic;
 
             return (object)sourceProperty != null &&
-                    sourceProperty.IsAutoPropertyWithGetAccessor &&
+                    sourceProperty.IsAutoProperty && sourceProperty.GetMethod is not null &&
```

Comment:
This will need future adjustment specific to semi auto properties. I didn't add a PROTOTYPE here since there is a one in the tests file already (`AssignReadOnlyOnlyPropertyInConstructor`)